.github/workflows/ci.yml

```diff
@@ -157,6 +157,23 @@ jobs:
           path: image.tar
           retention-days: 1
 
+      - name: Build the test Docker image
+        run: |
+          docker build . \
+            --tag $IMAGE_NAME:test \
+            --build-arg ENVIRONMENT=ci \
```

Comment:
the Dockerfile specifies `ARG ENVIRONMENT_SETTINGS`, while you use `ENVIRONMENT` here. I'd opt for the shorter one.